miio/device.py

```diff
@@ -283,5 +283,86 @@ def get_properties(
 
         return values
 
+    @command(
+        click.argument("properties", type=str, nargs=-1, required=True),
+    )
+    def test_properties(self, properties):
+        """Helper to test device properties."""
+
+        def ok(x):
+            click.echo(click.style(x, fg="green", bold=True))
+
+        def fail(x):
+            click.echo(click.style(x, fg="red", bold=True))
+
+        try:
+            model = self.info().model
+        except Exception as ex:
+            _LOGGER.warning("Unable to obtain device model: %s", ex)
+            model = "<unavailable>"
+
+        click.echo(f"Testing properties {properties} for {model}")
+        valid_properties = {}
+        for property in properties:
+            try:
+                click.echo(f"Testing {property}.. ", nl=False)
+                resp = self.get_properties([property])
+                # Handle responses with one-element lists
+                if isinstance(resp, list) and len(resp) == 1:
+                    resp = resp.pop()
+                value = valid_properties[property] = resp
+                if value is None:
+                    fail("None")
+                elif not value:
+                    fail("Empty response")
+                else:
+                    ok(f"{value} {type(value)}")
+            except Exception as ex:
+                _LOGGER.warning("Unable to request %s: %s", property, ex)
+
+        from pprint import pformat as pf
```

Comment:
Does `click` depends on `pprint`? Should we make sure `pprint` is available?

Reply:
This is just to pretty-print the final list with values, I'll move it on top of the file.

Reply:
Got it: pprint is part of the python core libraries and doesn't need to be installed/required.